src/commands/referral/referralNew.ts

```diff
@@ -0,0 +1,78 @@
+import { SlashCommandSubcommandBuilder } from '@discordjs/builders';
+import { getPrismaClient } from '../../clients';
+import { AutocompleteHandler, CommandHandler } from '../command';
+import { parseDate } from './parseDate';
+import { searchServices, services } from './services';
+
+export const data = new SlashCommandSubcommandBuilder()
+  .setName('new')
+  .setDescription('add new referral code/link for service')
+  .addStringOption((option) =>
+    option
+      .setName('service')
+      .setDescription(
+        'service to refer(type more than 3 characters to see suggestion)'
+      )
+      .setRequired(true)
+      .setAutocomplete(true)
+  )
+  .addStringOption((option) =>
+    option
+      .setName('link_or_code')
+      .setDescription('referral link or code')
+      .setRequired(true)
+  )
+  .addStringOption((option) =>
+    option
+      .setName('expiry_date')
+      .setDescription('when the code/link expired (MM/DD/YYYY)')
+      .setRequired(true)
+  );
+
+export const autocomplete: AutocompleteHandler = async (interaction) => {
+  const searchTerm = interaction.options.getString('service', true);
+  if (searchTerm.length < 4) return;
+
+  const options = searchServices(searchTerm);
+  interaction.respond(options);
+};
+
+export const execute: CommandHandler = async (interaction) => {
+  const code = interaction.options.getString('link_or_code', true);
+
+  const service = interaction.options.getString('service', true).toLowerCase();
+  const hasService = services.find(
+    (option) => option.toLowerCase() === service
+  );
+  if (!hasService)
+    return interaction.reply(
```

Comment:
anh biết cái này nhảy dòng do prettier, but wrap it with `{}` pls.

Reply:
:)) ok em ko de? y' no nhay? dong luon

Reply:
I never write shit like that ==!

Reply:
hence "anh biết cái này do prettier"